docs/Homebrew-brew-Maintainer-Guide.md

```diff
@@ -0,0 +1,102 @@
+
+If possible, PRs should also have:
+
+- Linear commit history (i.e. no merge commits in PR branches).
+- GPG-signed commits (see the private `ops` repository for instructions on setting this up).
+
+## Automatic approvals
+
+To ensure that each PR has the opportunity to be seen and reviewed by any other maintainers who wish
+to take a look, all PRs require an approval before they can be merged. However, not every PR is
+reviewed by another maintainer, and some PRs are urgent enough that they need to be merged without
+an approval by another maintainer.
+
+As a compromise between always needing a review and allowing maintainers to merge PRs they deem ready,
+the `Triage` CI job will ensure that PRs cannot be merged until they've been open for 24 hours
+(only counting hours that occur during the business week). After the triage period has expired, the
+CI job will show up as "passed" and `BrewTestBot` will approve the PR, allowing it to be merged.
+This gives all maintainers a reasonable opportunity to review every PR, but won't block any PR for lack
+of reviews.
+
+If the PR is urgent enough that it is necessary to bypass that 24 hour window, the `critical` label
+can be applied to the PR. When this label is applied, the `Triage` CI job will immediately be
+successful and `BrewTestBot` will approve the PR.
+
+## CI
+
+Every PR in `Homebrew/brew` runs a series of CI tests to try to prevent bugs from being introduced.
+**A PR _must_ have passing CI before it can be merged**.
+
+There are many checks that run on every PR. The following is a quick list of the various checks and what they represent:
+
+- `Vendor Gems / vendor-gems`: This is skipped except for dependabot PRs. It updates the RBI files to match
+  any new/changed dependencies. See [Type Checking With Sorbet](Typechecking.md) for more information about RBI files
+  and typechecking.
+- `Triage / review`: This verifies that the PR has been open for long enough.
+  See [above](#automatic-approvals) for more information about automatic approvals.
+- `codecov/patch` and `codecov/project`: These show the Codecov report for the PR.
+  See [below](#brew-tests-and-codecov) for more info about Codecov.
+- `CI / vendored gems (Linux)`: This checks whether there was a change to the vendored gems on Linux that needs to be
+  committed to the PR branch.
+- `CI / test default formula (Linux)`: This runs `brew test-bot` on Linux to ensure it still works as expected.
+- `CI / syntax`: This is run first to check whether the PR passes `brew style` and `brew typecheck`. If this job fails the
+  following jobs will not run.
+- `CI / tap syntax (Linux)`: This runs `brew style` and `brew audit` on all official taps
+  (note that although this has Linux in its name, it does check `Homebrew/homebrew-core`,
+  `Homebrew/linuxbrew-core` and all cask repos).
+- `CI / docker`: This builds and deploys a new Homebrew docker image.
+- `CI / test everything (macOS)`: This runs `brew tests` on macOS.
```

Comment:
It does a few more things, too?

Reply:
I don't think it does _much_ more (in terms of actual tests being run). It does run `brew update-tests` which is a separate command so I could add that. What else were you thinking should be added?

Reply:
@Rylan12 `brew readall`, `brew test-bot --only-formulae --test-default-formula`, `brew doctor` seem like the important bits.